### PR TITLE
Remove trailing empty lines on checklist import

### PR DESCRIPTION
**Pull request by no-lex**
Resolves a minor oversight I found where upon importation of a checklist, trailing empty lines will get their own checklist entry. This change removes those empty trailing entries during the loading process.

I decided not to remove empty lines surrounded by text, since those could have some use as a spacer. (It may be worth keeping the spaces they create while removing their corresponding checkbox)

Screenshot from 2024-09-07 23-47-17

### CODE REVIEW

File: lib/checklist_screen.dart [hunk 1]
```diff
@@ -50,6 +50,16 @@ class ChecklistScreenState extends State<ChecklistScreen> {
             return;
           }
           String name = lines.removeAt(0);
+          //remove trailing empty lines
+          while(!lines.isEmpty)
```
Comment (apps4av) [inline lib/checklist_screen.dart:54]: use isNotEmpty

File: lib/checklist_screen.dart [hunk 2]
```diff
@@ -50,6 +50,16 @@ class ChecklistScreenState extends State<ChecklistScreen> {
...
+          {
+            if(lines[lines.length - 1].length == 0) {
```
Comment (apps4av) [inline lib/checklist_screen.dart:56]: use isEmpty

File: lib/checklist_screen.dart [hunk 3]
```diff
@@ -50,6 +50,16 @@ class ChecklistScreenState extends State<ChecklistScreen> {
...
+                lines.removeAt(lines.length - 1);
+            }
```
Comment (apps4av) [inline lib/checklist_screen.dart:58]: I suggest running a for loop here instead of while to make it more readable.

**Review by apps4av — CHANGES_REQUESTED** [review]

File: lib/checklist_screen.dart [hunk 4]
```diff
@@ -50,6 +50,17 @@ class ChecklistScreenState extends State<ChecklistScreen> {
             return;
           }
           String name = lines.removeAt(0);
+          //remove trailing empty lines, back to front, until no lines remain or
```
Comment (apps4av) [inline lib/checklist_screen.dart:53]: I see you are removing trailing empty lines, why not remove leading empty lines as well?

**Review by apps4av — COMMENTED** [review]
Now I think more about it, why not remove all empty lines? Empty line may be used for spacing steps but its not a step. Going forward, clicking all steps will move to the next checklist. If some empty lines remain, then use has to check them too, which makes no sense.

**Comment by no-lex** [comment]
My thought was that those intermediate empty lines would remain but that the check tick itself would be disabled, which I was thinking of targeting in a separate PR. If this is infeasible though, I can just delete all empty lines.

**Review by apps4av — APPROVED** [review]

**Comment by apps4av** [comment]
For concise code, use map() function instead. These 20 lines could be condensed into 2 line. Approved.

**Pull request merged**